packages/grafana-data/src/dataframe/processDataFrame.ts

```diff
@@ -230,17 +230,19 @@ export function guessFieldTypeForField(field: Field): FieldType | undefined {
 }
 
 /**
- * @returns a copy of the series with the best guess for each field type
- * If the series already has field types defined, they will be used
+ * @returns A copy of the series with the best guess for each field type.
+ * If the series already has field types defined, they will be used, unless `guessDefined` is true.
+ * @param series The DataFrame whose field's types should be guessed
+ * @param guessDefined Whether to guess types of fields with already defined types
  */
-export const guessFieldTypes = (series: DataFrame): DataFrame => {
+export const guessFieldTypes = (series: DataFrame, guessDefined = false): DataFrame => {
```

Comment:
What is the usecase to force the guess again for CloudWatch? Where does the types come from if they need to be rechecked again?

Reply:
There's no type inference on the backend side of things, so (excluding the @timestamp field) all fields are returned as strings, which is an issue when we want to show numeric metrics. Could implement some kind of type inference in the go backend I suppose?

Reply:
I am more wondering if backend should be sending the types as string if that is not inferred and just hardcoded? That way we would not need to force anything on frontend? May not be super important thought could be changed later.

Reply:
Well, it's hardcoded because the values returned by cloudwatch are always strings, but also the field values need to be assigned _some_ type as required by the `data.Field` interface in go.